src/linear_solver_logs/solve_log_full_ma.jl

```diff
@@ -0,0 +1,352 @@
+   being used, a vector containing the values to be averaged (res_window), and an indicator of where 
+   the next value should be placed (idx).
+- `resid_hist::Vector{Float64}`, a structure that contains the moving average of the error proxy, 
+   typically the norm of residual or gradient of normal equations.
+- `iota_hist::Vector{Float64}`, a structure that contains the moving average of the error proxy, 
+   typically the norm of residual or gradient of normal equations.
+- `width_hist::Vector{Int64}`, data structure that contains the widths used in the moving average 
+   calculation at each iteration.
+- `resid_norm::Function`, a function that accepts a single vector argument and returns a
+    scalar. Used to compute the residual size.
+- `iterations::Int64`, the number of iterations of the solver.
+- `converged::Bool`, a flag to indicate whether the system has converged by some measure.
+- `true_res`, a boolean indicating if we want the true residual computed instead of approximate.
+- `dist_info::DistInfo`, a structure that stores information about the sub-Exponential distribution.
+# Constructors
+- Calling `LSLogMA()` sets `collection_rate = 1`,  `lambda1 = 1`,
+    `lambda2 = 30`, `resid_hist = Float64[]`, `iota_hist = Float64[]`, `width_hist = Int64[]`, 
+    `resid_norm = norm` (Euclidean norm), `iterations = -1`, `converged = false`, 
+    `sampler = LinSysVecRowDetermCyclic`, `max_dimension = 0`, `sigma2 = nothing`, `omega = nothing`,
+    `eta = 1`, and `true_res = false`. The user can specify their own values of lambda1, lambda2, sigma2, omega, and eta using 
+    key word arguments as inputs into the constructor.
+"""
+mutable struct LSLogMA <: LinSysSolverLog
+    collection_rate::Int64
+    ma_info::MAInfo
+    resid_hist::Vector{Float64}
+    iota_hist::Vector{Float64}
+    width_hist::Vector{Int64}
+    resid_norm::Function
+    iterations::Int64
+    converged::Bool
+    true_res::Bool
+    dist_info::DistInfo
+end
+
+LSLogMA(;
+        cr = 1, 
+        lambda1 = 1, 
+        lambda2 = 30, 
+        sigma2 = nothing, 
+        omega = nothing, 
+        eta = 1, 
+        true_res = false
+       ) = LSLogMA( cr,
+                    MAInfo(1, lambda2, 1, false, 1, zeros(lambda2)),
+                    Float64[], 
+                    Float64[], 
+                    Int64[],
+                    norm, 
+                    -1, 
+                    false,
+                    true_res,
+                    DistInfo(nothing, 0, 0, sigma2, omega, eta) 
+                  )
+
+#Function to update the moving average
+function log_update!(
+    log::LSLogMA,
+    sampler::LinSysSampler,
+    x::AbstractVector,
+    samp::Tuple,
+    iter::Int64,
+    A::AbstractArray,
+    b::AbstractVector
+)
+    if iter == 0 
+        # Check if it is a row or column method and record dimensions
+        if sum(supertype(typeof(sampler)) .<: [LinSysVecRowSampler, LinSysVecRowSketch, LinSysVecRowSelect]) > 1
```

Comment:
This is very pleasing code, but I think it needs more comment. You are checking to see which parent type of `sampler` does this fall under correct? Are `LinSysVecRowSampler`,... not just aliases of each other?

Reply:
Yeah they are just aliases. I have removed this.